pkg/controller/toolchainconfig/configuration.go

```diff
@@ -0,0 +1,61 @@
+package toolchainconfig
+
+import "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+
+type ToolchainConfig struct {
+	cfg *v1alpha1.ToolchainConfigSpec
+}
+
+func (c *ToolchainConfig) AutomaticApproval() *autoApproval {
+	return &autoApproval{c.cfg.Host.AutomaticApproval}
+}
+
+func (c *ToolchainConfig) Deactivation() *deactivation {
+	return &deactivation{c.cfg.Host.Deactivation}
+}
+
+type autoApproval struct {
+	approval v1alpha1.AutomaticApprovalCfg
+}
+
+func (a *autoApproval) IsEnabled() bool {
+	return getBool(a.approval.Enabled, false)
+}
+
+func (a *autoApproval) ResourceCapacityThresholdDefault() int {
+	return getInt(a.approval.ResourceCapacityThreshold.DefaultThreshold, 0)
+}
+
+func (a *autoApproval) ResourceCapacityThresholdSpecificPerMemberCluster() map[string]int {
+	return a.approval.ResourceCapacityThreshold.SpecificPerMemberCluster
+}
+
+func (a *autoApproval) MaxNumberOfUsersOverall() int {
+	return getInt(a.approval.MaxNumberOfUsers.Overall, 0)
+}
+
+func (a *autoApproval) MaxNumberOfUsersSpecificPerMemberCluster() map[string]int {
+	return a.approval.MaxNumberOfUsers.SpecificPerMemberCluster
+}
+
+type deactivation struct {
+	dctv v1alpha1.DeactivationCfg
+}
+
+func (d *deactivation) DeactivatingNotificationInDays() int {
+	return getInt(d.dctv.DeactivatingNotificationDays, 3)
+}
+
+func getBool(value *bool, defaultValue bool) bool {
+	if value != nil {
+		return *value
+	}
+	return defaultValue
+}
+
+func getInt(value *int, defaultValue int) int {
+	if value != nil {
+		return *value
+	}
+	return defaultValue
+}
```

Comment:
Let's move all relevant tests from the cache test to configuration_test.go.
In cache_test.go we need to test the cache itself. And in configuration_test.go we need to test the Configuration (wrapper) API & default values.

Reply:
Done in https://github.com/codeready-toolchain/host-operator/pull/438/commits/90af2f9b7c30cd88fe31d833c812ade8e60a2315